circle.yml

```diff
@@ -1,17 +1,84 @@
 version: 2
+
+defaults: &defaults
+  docker:
+    - image: circleci/node:6.12.3-browsers
+  working_directory: ~/axe-core
+
+restore_dependency_cache: &restore_dependency_cache
+  restore_cache:
+    keys:
+      - v1-npm-cache-{{ checksum "package.json" }}
+      - v1-npm-cache-
+
+set_npm_auth: &set_npm_auth
+  run: npm config set "//registry.npmjs.org/:_authToken" $NPM_AUTH
+
 jobs:
-  build:
-    docker:
-      - image: circleci/node:6.12.3-browsers
-    working_directory: ~/axe-core
+  # Fetch and cache dependencies.
+  dependencies:
+    <<: *defaults
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - npm-deps-{{ checksum "package.json" }}
+      - <<: *set_npm_auth
+      - <<: *restore_dependency_cache
       - run: npm install
       - save_cache:
-          key: npm-deps-{{ checksum "package.json" }}
+          key: v1-npm-cache-{{ checksum "package.json" }}
           paths:
             - node_modules
-      - run: npm test
\ No newline at end of file
+
+  # Run the test suite.
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_dependency_cache
+      - run: npm run test
+
+  # Release a "canary" version
+  canary_release:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *set_npm_auth
+      - <<: *restore_dependency_cache
+      - run: build/canary.js
+      - run: npm publish --tag=canary
```

Comment:
I think this needs to be `--tag=next`. That's certainly how we've done it so far, and it's what axe Coconut uses.

Reply:
Sure, will update. It'd be nice if we caught these things in the planning document rather than during implementation, but oh well 🙃